kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java

```diff
@@ -1223,20 +1225,87 @@ public void testFindTaskEvents() throws Exception {
 
             TaskSummary taskInstance = tasks.get(0);
 
+            TaskEventInstance expectedTaskEventInstance = TaskEventInstance
+                    .builder()
+                    .type(TaskEvent.TaskEventType.ADDED.toString())
+                    .processInstanceId(processInstanceId)
+                    .taskId(taskInstance.getId())
+                    .user(PROCESS_ID_USERTASK)
+                    .build();
+
             List<TaskEventInstance> events = taskClient.findTaskEvents(CONTAINER_ID, taskInstance.getId(), 0, 10);
             assertNotNull(events);
             assertEquals(1, events.size());
+            assertTaskEventInstance(expectedTaskEventInstance, events.get(0));
 
             // now let's start it
             taskClient.startTask(CONTAINER_ID, taskInstance.getId(), USER_YODA);
             events = taskClient.findTaskEvents(CONTAINER_ID, taskInstance.getId(), 0, 10);
             assertNotNull(events);
             assertEquals(2, events.size());
+            assertTaskEventInstance(expectedTaskEventInstance, events.get(0));
+            expectedTaskEventInstance.setType(TaskEvent.TaskEventType.STARTED.toString());
+            expectedTaskEventInstance.setUserId(USER_YODA);
+            assertTaskEventInstance(expectedTaskEventInstance, events.get(1));
         } finally {
             processClient.abortProcessInstance(CONTAINER_ID, processInstanceId);
         }
     }
 
+    @Test
+    public void testFindTaskEventsForNotExistingTask() {
+        try {
+            taskClient.findTaskEvents(CONTAINER_ID, -9999l, 0, 10);
+            fail("KieServicesException should be thrown complaining about task not found.");
+        } catch (KieServicesException e) {
```

Comment:
Assertions.assertThatThrownBy(() -> taskClient.findTaskEvents(CONTAINER_ID, -9999l, 0, 10))
                .isInstanceOf(KieServicesException.class)
                .hasMessageContaining("No task found with id");

As a point to counter your statement below that it would be confusing for others to employ AssertJ here.